core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1219,10 +1216,10 @@ class Log(@volatile private var _dir: File,
           fetchDataInfo = segment.read(startOffset, maxLength, maxPosition, minOneMessage)
           if (fetchDataInfo != null) {
             if (includeAbortedTxns)
-              fetchDataInfo = addAbortedTransactions(startOffset, segmentEntry, fetchDataInfo)
-          } else segmentEntryOpt = segments.higherEntry(baseOffset)
+              fetchDataInfo = addAbortedTransactions(startOffset, segment, fetchDataInfo)
+          } else segmentOpt = segmentsIterator.nextOption()
 
-          done = fetchDataInfo != null || segmentEntryOpt.isEmpty
+          done = fetchDataInfo != null || segmentOpt.isEmpty
```

Comment:
can we simply use `while (fetchDataInfo == null && segmentOpt.isDefined)` rather than `while (!done)`?

Reply:
Done.